homeassistant/components/fibaro/config_flow.py

```diff
@@ -83,3 +88,49 @@ async def async_step_user(
     async def async_step_import(self, import_config: ConfigType | None) -> FlowResult:
         """Import a config entry."""
         return await self.async_step_user(import_config)
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle reauthentication."""
+        self._reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by reauthentication."""
+        errors = {}
+
+        if user_input is not None and self._reauth_entry is not None:
+            new_data = self._reauth_entry.data | user_input
+            try:
+                info = await _validate_input(self.hass, new_data)
+            except FibaroConnectFailed:
+                errors["base"] = "cannot_connect"
+            except FibaroAuthFailed:
+                errors["base"] = "invalid_auth"
+            else:
+                self.hass.config_entries.async_update_entry(
+                    self._reauth_entry, data=new_data, unique_id=info["serial_number"]
+                )
+                self.hass.async_create_task(
+                    self.hass.config_entries.async_reload(self._reauth_entry.entry_id)
+                )
+                return self.async_abort(reason="reauth_successful")
+
+        username = (
+            self._reauth_entry.data[CONF_USERNAME]
+            if self._reauth_entry is not None
+            else None
+        )
```

Comment:
We shouldn't update the unique id them change accounts in reauth as the unique id will be wrong and the integration doesn't handle all the devices potentially changing. If it is a new entry they should setup a new one.

`self._reauth_entry` will never be None here.  You can add a positive assert to satisify mypy here

Reply:
Ok I understand, only password change should be possible. I changed that accordingly.